### PR TITLE
CKFVS Fix

### PR DESCRIPTION
**Pull request by isaiahvita**
P113014622

### CODE REVIEW

File: codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/CloudFrontKVSSigV4a.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+
+        var v4a = SigV4ATrait.builder()
+                .name(service.expectTrait(SigV4Trait.class).getName())
+                .build();
+
+        return model.toBuilder()
+                .addShape(
+                        service.toBuilder()
+                                .addTrait(v4a)
+                                .addTrait(new AuthTrait(SetUtils.of(SigV4ATrait.ID, SigV4Trait.ID)))
```
Comment (lucix-aws) [inline codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/CloudFrontKVSSigV4a.java:73]:
```suggestion
                                .addTrait(v4a)
                                // FUTURE: https://github.com/aws/smithy-go/issues/493
                                // we are keeping sigv4 at the end of this list (it will never be selected) as a stopgap to drive codegen of payload checksum routines
                                .addTrait(new AuthTrait(SetUtils.of(SigV4ATrait.ID, SigV4Trait.ID)))
```

File: codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/CloudFrontKVSSigV4a.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        if (!isCFKVSService(model, model.expectShape(settings.getService(), ServiceShape.class))) {
+            return;
+        }
+        runtimeClientPlugins.add(
+                RuntimeClientPlugin.builder()
+                    .configFields(
+                        ListUtils.of(
+                                ConfigField.builder()
+                                        .name(AwsSignatureVersion4aUtils.V4A_SIGNER_INTERFACE_NAME)
+                                        .type(SymbolUtils.createValueSymbolBuilder(
```
Comment (lucix-aws) [inline codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/CloudFrontKVSSigV4a.java:90]: nonblocking, remember that `SymbolUtils.buildPackageSymbol(String)` is now available (prefer usage via static import)

File: codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/CloudFrontKVSSigV4a.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+    @Override
+    public void writeAdditionalFiles(
+            GoSettings settings,
+            Model model,
+            SymbolProvider symbolProvider,
+            GoDelegator goDelegator
+    ) {
+
+        if (!isCFKVSService(model, model.expectShape(settings.getService(), ServiceShape.class))) {
```
Comment (lucix-aws) [inline codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/CloudFrontKVSSigV4a.java:119]: nonblocking

```suggestion
        if (!isCFKVSService(model, settings.getService(model))) {
```

File: codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/auth/BackfillSigV4ATrait.java [hunk 4]
```diff
@@ -31,7 +31,11 @@
 public class BackfillSigV4ATrait implements GoIntegration {
     private boolean isBackfillService(ServiceShape service) {
         final String sdkId = service.expectTrait(ServiceTrait.class).getSdkId();
-        return sdkId.equalsIgnoreCase("s3") || sdkId.equalsIgnoreCase("eventbridge");
```
Comment (lucix-aws) [inline codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/auth/BackfillSigV4ATrait.java:34]: Is this change still necessary?

**Review by lucix-aws — APPROVED** [review]
fixandship

**Review by wty-Bryant — APPROVED** [review]

File: codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/auth/BackfillSigV4ATrait.java (hunk 4 above)
Comment (isaiahvita) [reply codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/auth/BackfillSigV4ATrait.java]: nope
